docs/pages/reference/backends.mdx

```diff
@@ -172,6 +172,184 @@ teleport:
+of Teleport data:
+
+- Cluster state
+- Audit log events
+
+The PostgreSQL backend requires PostgreSQL 13 or later, and, for the cluster
+state only, the [`wal2json`](https://github.com/eulerto/wal2json) logical
+decoding plugin. The plugin is available in packages for all stable versions in
+the PostgreSQL [Apt](https://apt.postgresql.org/) and
+[Yum](https://yum.postgresql.org/) repositories for Debian- and RPM-based Linux
+distributions respectively, or it can be compiled following [the
+instructions](https://github.com/eulerto/wal2json#build-and-install) provided in
+its repository. The plugin is pre-installed with no extra steps to take in
+[Azure Database for
+PostgreSQL](https://azure.microsoft.com/en-us/products/postgresql).
+
+Teleport needs separate databases for the cluster state and the
+audit log, and it will attempt to create them if given permissions to do so; it
+will also set up the database schemas as needed, so we recommend giving the user
+ownership over the databases.
+
+The PostgreSQL backend for cluster state relies on the ability to use [logical
+decoding](https://www.postgresql.org/docs/current/logicaldecoding.html) to get a
+stream of changes from the database; because of that, the
+[`wal_level`](https://www.postgresql.org/docs/current/runtime-config-wal.html#GUC-WAL-LEVEL)
+parameter must be set to `logical` and
+[`max_replication_slots`](https://www.postgresql.org/docs/current/runtime-config-replication.html#GUC-MAX-REPLICATION-SLOTS)
+must be set to at least as many Teleport auth servers as you'll be running (a
+higher number is recommended, to account for network conditions). `wal_level`
+can only be set at server start, so it should be set in `postgresql.conf`:
+```
+# the default value for wal_level is replica
+wal_level = logical
+
+# the default value for max_replication_slots is 10
+max_replication_slots = 10
+```
+
+In addition, the database user must have the `initiating replication` [role
+attribute](https://www.postgresql.org/docs/current/role-attributes.html) (as
+well as `login privilege`, of course). In the `psql` shell:
+
+```
+postgres=# CREATE USER new_user WITH REPLICATION;
+CREATE ROLE
+
+postgres=# ALTER ROLE existing_user WITH LOGIN REPLICATION;
+ALTER ROLE
+```
+
+Since replication permissions allow for essentially full read access over the
+entire cluster (with a physical replication connection) or to all databases that
+the user can connect to, it's recommended to prevent the user from opening
+replication connections, or from connecting to databases other than the ones
+used for Teleport, if the PostgreSQL cluster is shared between Teleport and
+other applications.
+
+For convenience, Teleport will attempt to grant itself the `initiating
+replication` role attribute, if given the ability to do so (with a `superuser
+status` user, or similar privileged access as can be granted in managed
+PostgreSQL services such as Azure Databases for PostgreSQL or Amazon RDS for
+PostgreSQL).
```

Comment:
The parenthetical phrase is longer than the rest of the paragraph—I think we can restructure this to make it clearer. How does Teleport test whether it can grant itself the `initiating replication` role attribute?

Reply:
Simplified while also actually describing the purpose behind it in 0172239d3145327b25466a1a2946f6dcec123636, PTAL.